### ISSUE TITLE
Simulator not loading, also unable to cancel

### ISSUE THREAD

**Issue by chahud42** [mythridium/combat-simulator#30]
**Describe the bug**
The mod is pretty much broken for me at the moment. Simulator doesn't simulate.

**To Reproduce**
After loading my set and choosing a an enemy, upon clicking "simulate selected" the button gets stuck indefinitely on "Cancel (0/1)". When clicking cancel, the button gets stuck on "Cancelling". Same problem for "simulate all". You have to reload the app for it to change back, but it doesn't help.

**Expected behavior**
A simulation of the combat.

**Device:**
Mobile iOS 14.7.1

I just checked on desktop browser also (Opera) and it seems to be broken there too.

**Additional context**
It was working fine yesterday or maybe even earlier today, although it has been happening sporadically recently. Now the issue seems to persist. I don't _think_ I'm just impatient...a single simulation should be pretty quick. No errors n the dev console.

**Save**
[melvoridlesave_Philidor_3-23-2024_123740AM.txt](https://github.com/mythridium/combat-simulator/files/14730220/melvoridlesave_Philidor_3-23-2024_123740AM.txt)

**Mod List**
[Myth] Combat Simulator
SEMI Core
dev.Console
SEMI Auto Loot
SEMI Auto Eat
Mastery Virtual Levels
SEMI Auto Sell
SEMI Auto Slayer
SEMI Auto Farming
Mastery Pool Can Overflow
QuickShards (Disabled)
Bank Tab Values
ETA
Better Summoning Menu
Equipment Presets
Falterfire's Anti-Loitering
Rune Shop
Warriors Guild (Defenders)
Priority Miner
Better Offline Recap
Show Dungeon Completion Count
Upgrade All Potions
Mining Quarry
[Myth] Music
Abyssal Rift
Treasure Trails - Clue Scrolls
Speedup
Show Item Sources And Uses
[Bank] Item placeholder
Free Extra Bank Slots
AutoStoner
Kills in Combat

It's a lot but I don't think any of them are the problem, it has worked before with all of these all at once. That said, I am an idiot so there's that.

Thanks!

**Comment by mythridium** [comment]
Thanks for reporting this issue. The current version of combat simulator does not support content mods. This is something I have been trying to improve and there are some improvements coming in the next major combat simulator update which is lined up for release with the new melvor expansion coming out.

This was a compatibility issue between the Abyssal Rifts mod and the Combat Simulator. The simulator does not know about the existence of modded content and it was trying to pick up one of the abyssal rift shop purchases when importing your equipment set and resulted in an error being thrown when trying to simulate.  You must have purchased a shop upgrade from the mod recently which explains why it stopped working only recently.

I've released a fix so that this won't cause the simulator to break anymore.

Fixed in cc9ef6d0d6a52e2d743036bb5c61e5ed567c8049
